W1-M6-object-oriented/LineOfCreditAccount.cs

```diff
@@ -0,0 +1,27 @@
+namespace classes
+{
+    public class LineOfCreditAccount : BankAccount
+    {
+        public LineOfCreditAccount(string name, decimal initialBalance, decimal creditLimit) : base(name, initialBalance -creditLimit)
+        {
+        }
+
+        public override void PerformMonthEndTransactions()
+        {
+            if (Balance < 0)
+            {
+                var interest = -Balance * 0.07m;
+                MakeWithdrawal(interest, DateTime.Now, "Charge monthly interest");
+            }
+
+        }
+       /// <summary> 
+       /// I Do not understand what is occuring in this function  
+       ///</summary>
+        protected override Transaction? CheckWithdrawalLimit(bool isOverdrawn) =>
```

Comment:
You are checking the withdrawal limit and passing a isOverdrawn

If overdrawn , returning a new transaction dated now with a value of -20 with a comment of "Apply Overdraft fee", if not  returning null by the look of it.

You could write

```cs
if(isOverdrawn)
{
   var txn = new Transaction(-20,DateTime.Now,"Apply overdraft fee");
return txn;
}
else
{
    return null;
}
```